dashboard/test/controllers/api/v1/sections_controller_test.rb

```
@@ -95,4 +101,297 @@ class Api::V1::SectionsControllerTest < ActionController::TestCase
 
     assert_equal @course.id, json['courseId']
   end
+
+  test 'logged out cannot create a section' do
+    post :create
+    assert_response :forbidden
+  end
+
+  test 'student cannot create a section' do
+    sign_in @word_user_1
+    post :create
+    assert_response :forbidden
+  end
+
+  test 'teacher can create a section' do
+    sign_in @teacher
+    post :create
+    assert_response :success
+    refute_nil returned_section
+  end
+
+  test 'creating a section returns the section summary' do
+    sign_in @teacher
+    post :create
+
+    # See section_test.rb for tests covering the shape of the section summary.
+    assert_equal returned_section.summarize.with_indifferent_access,
+      returned_json.with_indifferent_access
+  end
+
+  test 'current user owns the created section' do
+    sign_in @teacher
+    post :create
+
+    assert_equal @teacher.name, returned_json['teacherName']
+    assert_equal @teacher, returned_section.user
+  end
+
+  test 'cannot override user_id during creation' do
+    sign_in @teacher
+    post :create, params: {user_id: (@teacher.id + 1)}
+    assert_response :success
+    # TODO: Better to fail here?
```

You'll see several of these - I've preserved the old API behavior in most cases, which is to completely ignore unwanted or malformed parameters and proceed with creating the section using default values.  I suspect it's better API design to reject the request with a 400 if a suspicious parameter is included.  Thoughts on that, and whether we should change it now as we port this API?

---

I like the idea of initially porting to match the existing API, and then making sure we schedule time to do a cleanup pass at some point where we maybe improve some things.

---

General Rails style is to filter to just the params you expect by defining [StrongParameters](http://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters). That effectively ignores unexpected params. There's no reason we couldn't also detect them and fail the request, but I don't think it's necessary.

---

Okay - I'm going to leave this as future cleanup then (strong params, rejecting requests that fail validation instead of going with defaults) except for the `login_type` case which it sounds like we ought to address for the current feature.